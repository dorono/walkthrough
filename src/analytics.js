export const trackPageView = path => window.ga('send', 'pageview', path);
