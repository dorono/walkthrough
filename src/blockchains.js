export const AVAILABLE_BLOCKCHAINS = {
    MAINNET: {
        label: 'Mainnet',
        value: 0,
        serviceName: 'mainnet',
        url: CONFIG.apiUrls.mainnet,
    },
    SHARED: {
        label: 'Shared Sandbox',
        value: 1,
        serviceName: 'shared',
        url: CONFIG.apiUrls.sharedSandbox,
    },
    PRIVATE: {
        label: 'Private Network',
        value: 2,
        serviceName: 'private_network',
        url: null,
    },
};

Object.freeze(AVAILABLE_BLOCKCHAINS);
