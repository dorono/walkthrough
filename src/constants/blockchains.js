export const AVAILABLE_BLOCKCHAINS = {
    PUBLIC: {
        label: CONFIG.blockchainNetwork,
        value: 0,
        serviceName: 'mainnet',
        url: CONFIG.gatewayUrls.publicNet,
    },
    SHARED: {
        label: 'Shared Sandbox',
        value: 1,
        serviceName: 'shared',
        url: CONFIG.gatewayUrls.sharedSandbox,
    },
    PRIVATE: {
        label: 'Private Network',
        value: 2,
        serviceName: 'private_network',
        url: null,
    },
};

Object.freeze(AVAILABLE_BLOCKCHAINS);
