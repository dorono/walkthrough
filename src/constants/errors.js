export const ERRORS = {
    CREDENTIALS: 'credentials',
    CUSTOM_CREDENTIALS_NOT_ALLOWED: 'custom-credentials-not-allowed',
    NETWORK: 'network',
    OTHER: 'other',
};

export const ERROR_MESSAGES = {
    AUTHENTICATION_FAILED: 'Authentication failed',
};

Object.freeze(ERRORS);
