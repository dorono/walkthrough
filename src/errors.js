export const ERRORS = {
    CREDENTIALS: 'credentials',
    CUSTOM_CREDENTIALS_NOT_ALLOWED: 'custom-credentials-not-allowed',
    NETWORK: 'network',
    OTHER: 'other',
};

Object.freeze(ERRORS);
