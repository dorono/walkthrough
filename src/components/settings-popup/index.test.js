import React from 'react';
import {shallow} from 'enzyme';
import APIConfig from 'utils/api-config';
import {AVAILABLE_BLOCKCHAINS} from 'blockchains';
import {ERRORS} from 'errors';
import {trackNotSuccessfulConnection} from 'analytics';
import SettingsPopup, {getDefaultErrors} from './index';

// Ask Jest to mock the analytics module.
jest.mock('../../analytics');

describe('SettingsPopup Component', () => {
    const noOp = jest.fn();
    const apiConfig = new APIConfig();

    it('should render without errors', () => {
        shallow(
            <SettingsPopup
                show
                onClose={noOp}
                onSubmit={noOp}
                defaultApiConfig={apiConfig}
                apiConfig={apiConfig}
            />,
        );
    });

    it('should render the popup', () => {
        const wrapper = shallow((
            <SettingsPopup
                show
                defaultApiConfig={apiConfig}
                onClose={noOp}
                onSubmit={noOp}
                apiConfig={apiConfig}
            />
        ));
        const modalBody = wrapper.find('ModalBody');
        expect(modalBody.length).toBe(1);
    });

    it('should have a Form', () => {
        const wrapper = shallow((
            <SettingsPopup
                show={false}
                defaultApiConfig={apiConfig}
                onClose={noOp}
                onSubmit={noOp}
                apiConfig={apiConfig}
            />
        ));
        expect(wrapper.find('Form').length).toBe(1);
    });

    it('checkbox should be disabled and set to true after selecting something different to Public network', () => {
        const wrapper = shallow((
            <SettingsPopup
                show
                defaultApiConfig={apiConfig}
                onClose={noOp}
                onSubmit={noOp}
                apiConfig={apiConfig}
            />
        ));
        wrapper.instance().handleDropdownChange(AVAILABLE_BLOCKCHAINS.SHARED);
        expect(wrapper.state('enableCredentialsCheckbox')).toBe(false);
        expect(wrapper.state('useCredentials')).toBe(true);
    });

    it('should track error type on api configuration validation', async () => {
        const settingsPopup = shallow(<SettingsPopup
            show
            defaultApiConfig={apiConfig}
            onClose={noOp}
            onSubmit={noOp}
            apiConfig={apiConfig}
        />);
        const p = Promise.resolve();
        settingsPopup.instance().handleSubmit();
        await p;
        expect(trackNotSuccessfulConnection).toHaveBeenCalledTimes(1);
        expect(trackNotSuccessfulConnection).toHaveBeenCalledWith(ERRORS.OTHER);
    });

    it('renderErrorMessage should return a valid error message', async () => {
        const settingsPopup = shallow(<SettingsPopup
            show
            defaultApiConfig={apiConfig}
            onClose={noOp}
            onSubmit={noOp}
            apiConfig={apiConfig}
        />);
        let errors = getDefaultErrors();

        errors[ERRORS.NETWORK] = true;
        settingsPopup.setState({errors});
        expect(settingsPopup.instance().renderErrorMessage().type.name)
            .toBe('NetworkErrorMessage');

        errors = getDefaultErrors();
        errors[ERRORS.CREDENTIALS] = true;
        settingsPopup.setState({errors});
        expect(settingsPopup.instance().renderErrorMessage().type.name)
            .toBe('CredentialsErrorMessage');

        errors = getDefaultErrors();
        errors[ERRORS.OTHER] = true;
        settingsPopup.setState({errors});
        expect(settingsPopup.instance().renderErrorMessage().type.name)
            .toBe('GenericErrorMessage');
    });
});
