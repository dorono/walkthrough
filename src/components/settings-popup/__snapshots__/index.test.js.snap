// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsPopup Component checkbox should be disabled and set to true after selecting something different to Public network 1`] = `
<Modal
  className="modalRoot defaultPosition"
  setModalSize={false}
  show={true}
>
  <ModalHeader
    className="modalHeader"
  >
    BLOCKCHAIN SETTINGS
  </ModalHeader>
  <ModalBody
    className="modalBody"
  >
    <_default
      className="messageBar"
      message={null}
      show={false}
      type="error"
    />
    <Form
      className="form"
    >
      <FormGroup
        className="formGroup"
        id="target-blockchain-select"
      >
        <Label
          className="label"
        >
          Target blockchain
        </Label>
        <OnClickOutside(Dropdown)
          arrowColor="blue"
          className="dropdown"
          disabled={false}
          eventTypes={
            Array [
              "mousedown",
              "touchstart",
            ]
          }
          excludeScrollbar={false}
          headerClassName="dropdownHeader formInput"
          onOptionClick={[Function]}
          options={
            Array [
              Object {
                "label": undefined,
                "serviceName": "mainnet",
                "url": "",
                "value": 0,
              },
              Object {
                "label": "Shared Sandbox",
                "serviceName": "shared",
                "url": "",
                "value": 1,
              },
              Object {
                "label": "Private Network",
                "serviceName": "private_network",
                "url": null,
                "value": 2,
              },
            ]
          }
          optionsClassName="dropdownOptions"
          outsideClickIgnoreClass="ignore-react-onclickoutside"
          preventDefault={false}
          selected={
            Object {
              "label": "Shared Sandbox",
              "serviceName": "shared",
              "url": "",
              "value": 1,
            }
          }
          selectedClassName="selectedOption"
          stopPropagation={false}
        />
        <div
          className="information"
        >
          <div
            className="checkboxContainer"
          >
            <Checkbox
              onChange={[Function]}
              readOnly={true}
              value={true}
            />
            Use my Harmony credentials
          </div>
        </div>
      </FormGroup>
      <div
        className="link"
      >
        <div
          className="questionMark"
        >
          ?
        </div>
        <_default
          text=" What is Harmony?"
          to="https://www.factom.com/products/harmony-connect"
        />
      </div>
      <div
        className="transition"
      >
        <FormGroup
          className="formGroup"
          key="app-id"
        >
          <Label
            className="label"
          >
            Harmony Application ID
          </Label>
          <Input
            className="input formInput"
            error={false}
            handleChange={[Function]}
            key="appId"
            name="appId"
            placeholder="Enter your App ID"
            type="text"
            value=""
          />
        </FormGroup>
        <FormGroup
          className="formGroup"
          key="app-key"
        >
          <Label
            className="label"
          >
            Harmony Application Key
          </Label>
          <Input
            className="input formInput"
            error={false}
            handleChange={[Function]}
            key="appKey"
            name="appKey"
            placeholder="Enter your App Key"
            type="password"
            value=""
          />
        </FormGroup>
      </div>
    </Form>
  </ModalBody>
  <ModalFooter
    className="modalFooter"
  >
    <Button
      className="button cancel"
      disabled={false}
      id="cancel-settings-popup"
      key="cancel"
      onClick={[Function]}
      title="CANCEL"
    />
    <Button
      className="button"
      disabled={false}
      key="submit"
      onClick={[Function]}
      title="SUBMIT"
    />
  </ModalFooter>
</Modal>
`;
